### ISSUE TITLE
Suggested API update

### ISSUE THREAD

**Issue by Quantumplation** [antibody-cardano/translucent#41]
Given [recent issues](https://github.com/input-output-hk/nami/issues/925), I'd like to propose updating the surface API of translucent to prevent accidental mistakes.

Currently, [payToContract](https://github.com/antibody-cardano/translucent/blob/d061f3839eabf464a984676442478d5b7c9acaf0/packages/translucent/src/translucent/tx.ts#L340-L344) prevents sending to a smart contract [if no datum is attached](https://github.com/antibody-cardano/translucent/blob/d061f3839eabf464a984676442478d5b7c9acaf0/packages/translucent/src/translucent/tx.ts#L349-L353).

This is super useful to prevent accidental mistakes, but occasionally people will accidentally use [the payToAddressWithData](https://github.com/antibody-cardano/translucent/blob/d061f3839eabf464a984676442478d5b7c9acaf0/packages/translucent/src/translucent/tx.ts#L273-L277) method instead, which bypasses this check.

If we made this method private, and added `payToNativeScriptWithData` and `payToWalletAddressWithData` that, respectively, required you to pass the native address and required you to pass a non-script address, this could solve some accidental use of the library.

**Comment by micahkendall** [comment]
Makes sense but generally we're keeping the lucid apis for compatibility, with intent to move away entirely at some point. Later variations can have such considerations.
